Extensions/NotificationService/NotificationService.swift

```diff
@@ -24,52 +24,36 @@ class NotificationService: UNNotificationServiceExtension {
         let content = request.content.mutableCopy() as! UNMutableNotificationContent
 
         if self.profile == nil {
-            self.profile = ExtensionProfile(localName: "profile")
+            self.profile = BrowserProfile(localName: "profile")
         }
 
         guard let profile = self.profile else {
             self.didFinish(with: .noProfile)
             return
         }
 
-        let queue = profile.queue
-        let display = SyncDataDisplay(content: content, contentHandler: contentHandler, tabQueue: queue)
+        let display = SyncDataDisplay(content: content, contentHandler: contentHandler)
         self.display = display
-        profile.syncDelegate = display
 
         let handler = FxAPushMessageHandler(with: profile)
 
         handler.handle(userInfo: userInfo).upon { res in
-            guard res.isSuccess, let events = res.successValue, let firstEvent = events.first else {
+            guard res.isSuccess, let event = res.successValue else {
```

Comment:
So now we only have one event at a time? Just wanna understand the context of this change.

Reply:
Yes! and indeed some context here would have been helpful for the review,

In the past, the `handlePush` API could retrieve multiple events (in practice, multiple tabs in case some were accidentally lost)

However, that causes a bit of an unfortunate API. `handlePush` would get your missing tabs, put both them in the tab queue, but only show you the push notification for one.

 Logically that seems okay, but, because the notification service is running on an app extension it could race with the app when the app polls for commands, making it possible for a user to receive duplicate send tabs. To make easy to see the race:
- Notification service loads tabs into tab queue, and updates last-seen-tab-index
- Meanwhile, main process pollsCommands with the old last-seen-tab-index
- main process opens the tabs that is the result of pollCommands
- main process also opens the tabs in the tab queue, which duplicate the tabs it opened earlier



Now, the Notification service does not try to recover any missing tabs, and instead transforms the encrypted payload into a single notification to display to the user, then lets the app itself poll for commands to catch any missing tabs. It simplifies the job of the notification service to be: "Convert a push payload into something we want to show the user" and removes the side-effects